### PR TITLE
Additional fixes to 0/1D tests

### PR DESCRIPTION
**Pull request by Abdurrahheem**
This has additional fixes requited for 0/1D tests. 

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/test/test_layers_1d.cpp [hunk 1]
```diff
@@ -42,8 +42,8 @@ TEST_P(Layer_1d_Test, Scale)
 
     cv::Mat output_ref = input.mul(weight);
     runLayer(layer, inputs, outputs);
-
-    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    ASSERT_EQ(1, outputs.size());
+    ASSERT_EQ(shape(outputs[0]), shape(output_ref));
```
Comment (fengyuentau) [inline modules/dnn/test/test_layers_1d.cpp:46]: Reference should be put in the first parameter and real output goes to the second.

See https://github.com/opencv/opencv/pull/25231#discussion_r1555470965.

File: modules/dnn/test/test_layers_1d.cpp [hunk 2]
```diff
@@ -430,16 +437,17 @@ INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Softmax_Test, Combine(
...
+    auto tup = get<0>(GetParam());
+    int axis = get<0>(tup);
+    std::vector<int> input_shape = get<1>(tup);
     std::string opr = get<1>(GetParam());
 
     LayerParams lp;
     lp.type = "Scatter";
     lp.name = "addLayer";
-    lp.set("axis", 0);
+    lp.set("axis", axis);
```
Comment (fengyuentau) [inline modules/dnn/test/test_layers_1d.cpp:450]: `axis = input_shape.size() - 1` should also work.

File: modules/dnn/test/test_layers_1d.cpp (hunk 1 above)
Comment (Abdurrahheem) [reply modules/dnn/test/test_layers_1d.cpp:46]: Done

**Comment by opencv-alalek** [comment]
> [Abdurrahheem](https://github.com/Abdurrahheem) added this to the [5.x](https://github.com/opencv/opencv/milestone/57) milestone

Why do you create a new milestone?

**Comment by Abdurrahheem** [comment]
> > [Abdurrahheem](https://github.com/Abdurrahheem) added this to the [5.x](https://github.com/opencv/opencv/milestone/57) milestone
> 
> Why do you create a new milestone?

accidentally, sorry

File: modules/dnn/test/test_layers_1d.cpp [hunk 3]
```diff
@@ -42,107 +42,139 @@ TEST_P(Layer_1d_Test, Scale)
...
 INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Gather_1d_Test, Combine(
-/*input blob shape*/    Values(0, 1, 2, 3),
-/*operation*/           Values(0, 1)
+/*input blob shape*/    testing::Values(
+                                std::vector<int>({}),
+                                std::vector<int>({1}),
+                                std::vector<int>({1, 4}),
+                                std::vector<int>({4, 4})
+                                ),
+/*operation*/           testing::Values(0, 1)
```
Comment (dkurt) [inline modules/dnn/test/test_layers_1d.cpp:97]:
```suggestion
/*axis*/           testing::Values(0, 1)
```

**Review by dkurt — APPROVED** [review]

**Comment by dkurt** [comment]
Please fix failed CUDA test: https://github.com/opencv/opencv/actions/runs/9075787020/job/24937114379?pr=25487

**Review by dkurt — APPROVED** [review]

**Pull request merged**